python/pyspark/sql/udf.py

```diff
@@ -449,11 +445,10 @@ def func(*args: Any, **kwargs: Any) -> Any:
                     return profiler.profile(f, *args, **kwargs)
 
                 func.__signature__ = inspect.signature(f)  # type: ignore[attr-defined]
+                resultId = sc._jvm.PythonSQLUtils.nextExprId()
```

Comment:
We can make this diff a bit smaller if we want to. We can still use the old method, as long as we can get rid of expr()...